### ISSUE TITLE
Adding Transparency to the Transpiler

### ISSUE THREAD

**Issue by kaelynj** [Qiskit/documentation#1046]
### Describe the new content you are requesting.

From the discussion in #878, I've identified a few topics for new pages/edits to other pages that I will work on adding in to the transpile section.  I'll keep track of the list of changes using this issue and will expand the list as needed.  Feel free to add suggestions or comments.

- [ ] Edit subsection on the `Target` object on the _Representing Quantum Computers_ page.  Make it more explicit what the `Target` object and what it's used for.
- [ ] Add page on `UnitarySynthesis`, with an explanation of the Quantum Shannon Decomposition and the multiplexor formalism.
- [ ] Edit the `qiskit.synthesis` [overview docstring](https://docs.quantum.ibm.com/api/qiskit/synthesis) to make more clear what the module is designed for (from what I understand it's the "engine" powering some/all of the transformation passes, among other things).
- [ ] Reorganize the items on the left sidebar of the `qiskit.transpiler.passes` API docs to match the section ordering found in its [overview page](https://docs.quantum.ibm.com/api/qiskit/transpiler_passes) (@Eric-Arellano is this possible by making a custom ordering for this part of the API docs?  Or would the API docs themselves need to be edited?)

### If this new content request is accepted, do you want to write the content?

I will write a draft of the proposed content

**Comment by Eric-Arellano** [comment]
> Reorganize the items on the left sidebar of the qiskit.transpiler.passes API docs to match the section ordering found in its [overview page](https://docs.quantum.ibm.com/api/qiskit/transpiler_passes) (@Eric-Arellano is this possible by making a custom ordering for this part of the API docs? Or would the API docs themselves need to be edited?)

It's not possible without adding substantial complexity to the docs generation script. If we weren't to make changes to the API script, then we'd need to reorganize the APIs to live in submodules. But that changes the actual import paths for end users in a bad way, so it's a non-starter.

If this is really important to change the left sidebar, we can consider it. But I'd bias against it since the script is already pretty complex.

**Comment by kaelynj** [comment]
> It's not possible without adding substantial complexity to the docs generation script. If we weren't to make changes to the API script, then we'd need to reorganize the APIs to live in submodules. But that changes the actual import paths for end users in a bad way, so it's a non-starter.
>
> If this is really important to change the left sidebar, we can consider it. But I'd bias against it since the script is already pretty complex.

Yeah I figured this might be the case.  From a user perspective, it's quite unclear how to navigate these API pages unless you go directly to the `Overview` page where it's TOC groups the related pages together.  Ordering them into sections instead of one long alphabetical list of pages would make navigating to particular types of passes significantly easier.

I totally understand this would be tricky, but I think it's an important enough quality of life improvement that we should consider it.

![transpilerPasses_ApiPage](https://github.com/Qiskit/documentation/assets/43348706/e47973c6-361c-4c27-b6d4-d73a0ee55612)
